app/assets/javascripts/alongslide/state.coffee

```diff
@@ -0,0 +1,30 @@
+#
+# state.coffee: update browser history
+#
+# Copyright 2013 Canopy Canopy Canopy, Inc.
+#
+class Alongslide::State
+
+  panelNames   : {}
+  documentTitle: ''
+  hash         : '#'
+
+  constructor: (options = {}) ->
+    History = window.History
+    if !History.enabled then return false
+    @documentTitle = document.getElementsByTagName('title')[0].innerHTML
+    @panelNames    = options.panelNames
+
+  rewindStateIndex:(state) ->
+    state.index -= 1
+    @update(state);
+
+  update:(state)->
+    if @panelNames[state.index] and state.index > -1
```

Comment:
I ask because it appears that `panelNames` is only updated in `buildPanel` in the `Layout`, so would apply to panels and not sections (which are created, along with their backgrounds, in `renderSection()` in `Layout`). Haven't tested the code, lemme know if I'm misunderstanding!